python/artm/model.py

```diff
@@ -0,0 +1,723 @@
+        topic_names = [topic_name for topic_name in theta_matrix[0].topic_name]
+        retval = DataFrame(data=theta_matrix[1].transpose(),
+                           columns=document_ids,
+                           index=topic_names)
+
+        return retval
+
+    def find_theta(self, batches=None, collection_name=None,
+                   data_path='', data_format='batches'):
+        """ArtmModel.find_theta() --- find Theta matrix for new documents
+
+        Args:
+          collection_name (str): the name of text collection (required if
+          data_format == 'bow_uci'), default=None
+          batches (list of str): list of file names of batches to be processed;
+          if not None, than data_format should be 'batches'; format '*.batch',
+          default=None
+          data_path (str):
+          1) if data_format == 'batches' =>
+          folder containing batches and dictionary;
+          2) if data_format == 'bow_uci' =>
+          folder containing docword.txt and vocab.txt files;
+          3) if data_format == 'vowpal_wabbit' => file in Vowpal Wabbit format;
+          4) if data_format == 'plain_text' => file with text;
+          default=''
+          data_format (str): the type of input data;
+          1) 'batches' --- the data in format of BigARTM;
+          2) 'bow_uci' --- Bag-Of-Words in UCI format;
+          3) 'vowpal_wabbit' --- Vowpal Wabbit format;
+          4) 'plain_text' --- source text;
+          default='batches'
+
+        Returns:
+          pandas.DataFrame: (data, columns, rows), where:
+          1) columns --- the ids of documents, for which the Theta
+          matrix was requested
+          2) rows --- the names of topics in topic model, that was
+          used to create Theta
+          3) data --- content of Theta matrix.
+        """
+        if collection_name is None and data_format == 'bow_uci':
+            raise IOError('ArtmModel.find_theta(): No collection name was given')
+
+        if not data_format == 'batches' and batches is not None:
+            raise IOError("ArtmModel.find_theta(): batches != None require" +
+                          "data_format == batches")
+
+        if not self._initialized:
+            raise RuntimeError("ArtmModel.find_theta(): Model does not exist yet. Use " +
+                               "ArtmModel.initialize()/ArtmModel.fit_*()")
+
+        target_folder = data_path
+        if not data_format == 'batches':
+            target_folder = tempfile.mkdtemp()
+        try:
+            batches_list = _parse_collection_inline(target_folder,
+                                                    data_path,
+                                                    data_format,
+                                                    'find_theta',
+                                                    collection_name,
+                                                    batches,
+                                                    batch_size=1000)
```

Comment:
Always name arguments when the list of them is long enough (number of arguments >= 3 then use names).
This prevents from an error when you add new arguments to the function.